api/C/include/csi_ll_pmp.h

```diff
@@ -0,0 +1,237 @@
+
+/*
+ * Set PMP entry address and size need to be controlled by index.
+ *
+ * This function only can be called in M-mode. If it is called
+ * in other mode, it may return an error or cause carsh. When this function is called
+ * with success, it will not be effective untill the core has been exit M-mode.
+ *
+ * @param idx: The index of PMP config entries
+ * @param base_addr: The base address of this entry need to be configured
+ * @param size: The size of this entry need to be configured
+ * @return : Status code
+ */
+csi_status_t csi_pmp_set_address(unsigned int idx, unsigned long base_addr,
+                                unsigned long size);
+
+/*
+ * Get PMP entry address and size been controlled by index.
+ *
+ * This function only can be called in M-mode. If it is called
+ * in other mode, it may return an error or cause carsh. When this function is called
+ * with success, it will not be effective untill the core has been exit M-mode.
+ *
+ * @param idx: The index of PMP config entries
+ * @param base_addr: A pointer to the buffer which is used to output the base address
+ *                    of this range
+ * @param size: A pointer to the buffer which is used to output the size of this entry
+ * @return : Status code
+ */
+csi_status_t csi_pmp_get_address(unsigned int idx, unsigned long *base_addr,
+                                unsigned long *size);
+
+/*
+ * Set PMP entry address mode by index.
+ *
+ * This function only can be called in M-mode. If it is called
+ * in other mode, it may return an error or cause carsh. When this function is called
+ * with success, it will not be effective untill the core has been exit M-mode.
+ *
+ * @param idx: The index of PMP config entries
+ * @param mode: The address mode of PMP config entries
+ * @return : Status code
+ */
+csi_status_t csi_pmp_set_address_mode(unsigned int idx, csi_addr_matching_t mode);
+
+/*
+ * Get PMP entry address mode by index.
+ *
+ * This function only can be called in M-mode. If it is called
+ * in other mode, it may return an error or cause carsh. When this function is called
+ * with success, it will not be effective untill the core has been exit M-mode.
+ *
+ * @param idx: The index of PMP config entries
+ * @param mode: A pointer to the buffer which is used to output the address mode of
+ *              this entry
+ * @return : Status code
+ */
+csi_status_t csi_pmp_get_address_mode(unsigned int idx, csi_addr_matching_t *mode);
+
+/*
+ * Set PMP enctry excecutable flag by index.
+ *
+ * This function only can be called in M-mode. If it is called
+ * in other mode, it may return an error or cause carsh. When this function is called
+ * with success, it will not be effective untill the core has been exit M-mode.
+ *
+ * @param idx: The index of PMP config entries
+ * @param flag_x: The excecutable flag of PMP config entries.
+ * @return : Status code
+ */
+csi_status_t csi_pmp_set_executable(unsigned int idx, bool flag_x);
```

Comment:
`csi_pmp_set/get_executable/writable/readable/lock` maybe can be set as one such as csi_pmp_set/get to set the pmp attributes.